### PR TITLE
Metrics api unsafessl

### PR DESCRIPTION
**Pull request by wymangr**
Add unsafeSsl paramater to skil certificate validation when connecting over HTTPS

### Checklist

- [x] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Fixes #3728 
Fixes https://github.com/kedacore/keda/issues/3821

[Relates to #972](https://github.com/kedacore/keda-docs/pull/972)

### CODE REVIEW

**Review by JorTurFer — COMMENTED** [review]
Looking good, only a suggestion inline.
We need to have all the commits signed for merging PRs (DCO check), could you sign yor commits? https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#i-didnt-sign-my-commit-now-what

File: CHANGELOG.md [hunk 1]
```diff
@@ -57,6 +57,7 @@ To learn more about active deprecations, we recommend checking [GitHub Discussio
 - **GCP Storage Scaler:** Add prefix and delimiter support ([#3756](https://github.com/kedacore/keda/issues/3756))
 - **Prometheus Scaler:** Introduce skipping of certificate check for unsigned certs ([#2310](https://github.com/kedacore/keda/issues/2310))
 - **Event Hubs Scaler:** Support Azure Active Direcotry Pod & Workload Identity for Storage Blobs ([#3569](https://github.com/kedacore/keda/issues/3569))
+- **Metrics API Scaler:** Add unsafeSsl paramater to skil certificate validation when connecting over HTTPS ([#3728](https://github.com/kedacore/keda/discussions/3728))
```
Comment (JorTurFer) [inline CHANGELOG.md:60]: I have just noticed a typo here, to `skil` instead of `skip`

Comment (JorTurFer) [inline CHANGELOG.md:60]:
```suggestion
- **Metrics API Scaler:** Add unsafeSsl paramater to skip certificate validation when connecting over HTTPS ([#3728](https://github.com/kedacore/keda/discussions/3728))
```

File: pkg/scalers/metrics_api_scaler_test.go [hunk 2]
```diff
@@ -67,6 +67,10 @@ var testMetricsAPIAuthMetadata = []metricAPIAuthMetadataTestData{
 	{map[string]string{"url": "http://dummy:1230/api/v1/", "valueLocation": "metric", "targetValue": "42", "authMode": "bearer"}, map[string]string{"token": "bearerTokenValue"}, false},
 	// fail bearerAuth without token
 	{map[string]string{"url": "http://dummy:1230/api/v1/", "valueLocation": "metric", "targetValue": "42", "authMode": "bearer"}, map[string]string{}, true},
+	// success unsafeSsl true
+	{map[string]string{"url": "http://dummy:1230/api/v1/", "valueLocation": "metric", "targetValue": "42", "unsafeSsl": "true"}, map[string]string{}, false},
+	// success unsafeSsl false
+	{map[string]string{"url": "http://dummy:1230/api/v1/", "valueLocation": "metric", "targetValue": "42", "unsafeSsl": "false"}, map[string]string{}, false},
```
Comment (zroubalik) [inline pkg/scalers/metrics_api_scaler_test.go:73]: Just a nit, please add some invalid value (non boolean) test for this param, thanks 🙏

**Review by zroubalik — CHANGES_REQUESTED** [review]

**Review by zroubalik — APPROVED** [review]

**Pull request merged**